src/marketplace.js

```diff
@@ -0,0 +1,321 @@
+            let fileBuffer = null;
+            if (this.rootKeyPairPath !== false) {
+                fileBuffer = await readFile(this.rootKeyPairPath);
+            }
+            if (fileBuffer) {
+                const wif = fileBuffer.toString('utf-8');
+                this.rootKeyPair = ECPair.fromWIF(wif);
+                source = 'fs:' + this.rootKeyPairPath;
+            } else {
+                this.rootKeyPair = ECPair.makeRandom();
+                if (this.rootKeyPairPath !== false) {
+                    await saveFile(this.rootKeyPairPath, this.rootKeyPair.toWIF());
+                }
+                source = 'random';
+            }
+            const buffer = this.rootKeyPair.getPublicKeyBuffer();
+            this.pubKeyMultihash = await this.chluIpfs.instance.vendor.storePublicKey(buffer);
+            this.chluIpfs.pin(this.pubKeyMultihash);
+            // TODO: request pin?
+        }
+        return {
+            keyPair: this.rootKeyPair,
+            pubKeyMultihash: this.pubKeyMultihash,
+            source
+        }; 
+    }
+
+    /**
+     * Get the list of vendors that is registered in this
+     * marketplace. The ID used to identify each vendor
+     * is the multihash of their public key, which can be
+     * used to get the pub key from IPFS.
+     * 
+     * @returns {Array.<string>}
+     * @memberof Marketplace
+     */
+    async getVendorIDs() {
+        await this.start();
+        return await this.db.getVendorIDs();
+    }
+
+    /**
+     * @typedef {Object} Vendor
+     * @property {string} vPubKeyMultihash the multihash of the vendor key (aka Vendor ID)
+     * @property {string} vmPubKeyMultihash the multihash of the vendor-marketplace key
+     * @property {string} mSignature hex encoded signature of the marketplace for the vendor-marketplace key
+     * @property {string} vSignature hex encoded signature of the vendor for the vendor-marketplace key
+     */
+
+    /**
+     * Get public data about a specific vendor
+     * 
+     * @param {string} id the multihash of this vendor's pub key, aka their ID
+     * @returns {Promise<Vendor>}
+     * @memberof Marketplace
+     */
+    async getVendor(id) {
+        await this.start();
+        const v = await this.db.getVendor(id);
+        if (v) {
+            // the full keypair is omitted intentionally
+            // so it does not get out by mistake
+            return {
+                mSignature: v.mSignature,
+                vPubKeyMultihash: v.vPubKeyMultihash,
+                vSignature: v.vSignature,
+                vmPubKeyMultihash: v.vmPubKeyMultihash
+            };
+        } else {
+            throw new Error('Vendor with key ' + id + ' is not registered');
+        }
+    }
+
+    /**
+     * Get the vendor-marketplace key pair for a vendor
+     * 
+     * @param {string} vendorId the vendor's public key multihash
+     * @returns {Promise<ECPair>}
+     * @memberof Marketplace
+     */
+    async getVMKeyPair(vendorId) {
```

Comment:
What if it is not found? Return 404, I would think.

Same debate as the earlier comment.